src/commcare_cloud/commands/ansible/run_module.py

```diff
@@ -197,3 +193,30 @@ def run(self, args, unknown_args):
         args.quiet = True
         del args.shell_command
         return RunAnsibleModule(self.parser).run(args, unknown_args)
+
+
+class Ping(CommandBase):
+    command = 'ping'
+    help = 'Ping specified or all machines to see if they have been provisioned yet.'
+
+    arguments = (
+        shared_args.INVENTORY_GROUP_ARG,
+        Argument('--silence-warnings', action='store_true',
+                 help="Silence shell warnings (such as to use another module instead)"),
+    ) + NON_POSITIONAL_ARGUMENTS
+
+    def modify_parser(self):
+        RunAnsibleModule(self.parser).modify_parser()
+
+    def run(self, args, unknown_args):
+        ping_shell_command = 'echo {{ inventory_hostname }}'
+
+        args.module = 'shell'
+        if args.silence_warnings:
+            args.module_args = 'warn=false ' + ping_shell_command
+        else:
+            args.module_args = ping_shell_command
+        args.skip_check = True
+        args.quiet = True
+        del ping_shell_command
```

Comment:
this line doesn't do anything, to my knowledge, but looks like it was maybe done in analogy to e.g. `del args.shell_command` above? The purpose of _that_ line is to remove an unnecessary argument from args, since it's (okay, somewhat sloppily, I admit) being repurposed right after as the `args` for `RunAnsibleModule`.

Reply:
obe